### PR TITLE
O3-1157: Added field to search and select patient identifiers from implementer tools

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
I have added a search field for searching and selecting a patient-identifier-type's UUID, similar to searching a concept UUID/ person-attribute-type UUID. Along with that, I've added the UUID validator for the field.

image

## Related Issue
https://issues.openmrs.org/browse/O3-1157

### CODE REVIEW

File: packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    setActivePatientIdentifierTypeUuid(patientIdentifierType.uuid);
+    setPatientIdentifierTypeUuid(patientIdentifierType.uuid);
+    setSearchTerm("");
+  };
+
+  const handleSearchTermChange = (evt) => setSearchTerm(evt.target.value);
+
+  const filteredResults: Array<PatientIdentifierType> | undefined =
+    useMemo(() => {
+      if (!isLoading && searchTerm && searchTerm !== "") {
```
Comment (ZacButko) [inline packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx:44]: "" is considered false, so `searchTerm` and `searchTerm !== ""` is redundant

File: packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import React, { useState, useMemo } from "react";
+import uniqueId from "lodash-es/uniqueId";
```
Comment (ZacButko) [inline packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx:2]: Is this the UUID library that we use site-wide?

**Review by ZacButko — APPROVED** [review]

File: packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx:44]: Oh yes, thanks, I'll just update it.

File: packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx (hunk 2 above)
Comment (vasharma05) [reply packages/apps/esm-implementer-tools-app/src/configuration/interactive-editor/value-editors/patient-identifier-type-search.tsx:2]: No, this is just for generating a unique id for the Input field.
The site-wide application is UUID version 4, afaik.

**Pull request merged**